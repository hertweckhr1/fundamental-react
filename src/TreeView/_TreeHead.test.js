import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import TreeHead from './_TreeHead';
import TreeView from './TreeView';

// mock shortid for snapshot testing
jest.mock('shortid', () => {
    let id = 1;

    return {
        generate: () => id++
    };
});

describe('<TreeHead />', () => {
    const treeHead = (
        <TreeHead>
            <TreeView.Col>Column Header 1</TreeView.Col>
            <TreeView.Col>Column Header 2</TreeView.Col>
            <TreeView.Col>Column Header 3</TreeView.Col>
            <TreeView.Col>Column Header 4</TreeView.Col>
        </TreeHead>
    );

    test('create tree head component', () => {
        let component = renderer.create(treeHead);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('open all tree from header', () => {
        const wrapper = mount(treeHead);

        wrapper
            .find('button.fd-tree__control')
            .at(0)
            .simulate('click');

        // Check that all opened
        wrapper.find('ul.fd-tree__group').forEach((node) => {
            expect(node.hasClass('is-hidden')).toBeFalsy();
            expect(node.prop('aria-hidden')).toBeFalsy();
        });
    });

    test('closed all tree from header when set to open', () => {
        const wrapper = mount(treeHead);

        wrapper
            .find('button.fd-tree__control')
            .at(0)
            .simulate('click');

        // Check that all opened
        wrapper.find('ul.fd-tree__group').forEach((node) => {
            expect(node.hasClass('is-hidden')).toBeFalsy();
            expect(node.prop('aria-hidden')).toBeFalsy();
        });

        wrapper
            .find('button.fd-tree__control')
            .at(0)
            .simulate('click');

        // Check that all closed
        wrapper.find('ul.fd-tree__group').forEach((node) => {
            expect(node.hasClass('is-hidden')).toBeTruthy();
            expect(node.prop('aria-hidden')).toBeTruthy();
        });
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the TreeHead component', () => {
            const element = mount(
                <TreeHead
                    data-sample='Sample' />);

            expect(
                element.find('.fd-tree--header').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the TreeHead component', () => {
            const element = mount(
                <TreeHead
                    data-sample='Sample' />);

            expect(
                element.find('.fd-tree--header').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the TreeHead component\'s button element', () => {
            const element = mount(
                <TreeHead
                    buttonProps={{
                        'data-sample': 'Sample'
                    }}>
                    <TreeView.Col />
                </TreeHead>
            );

            expect(
                element.find('.fd-tree__control').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
