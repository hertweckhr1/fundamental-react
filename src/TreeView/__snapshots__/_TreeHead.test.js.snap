// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeHead /> create tree head component 1`] = `
<div
  className="fd-tree fd-tree--header"
>
  <div
    className="fd-tree__row fd-tree__row--header"
  >
    <div
      className="fd-tree__col fd-tree__col--control"
    >
      <div>
        <button
          aria-label="expand all"
          className="fd-tree__control"
        />
        Column Header 1
      </div>
    </div>
    <div
      className="fd-tree__col"
    >
      Column Header 2
    </div>
    <div
      className="fd-tree__col"
    >
      Column Header 3
    </div>
    <div
      className="fd-tree__col"
    >
      Column Header 4
    </div>
  </div>
</div>
`;
